tests.json

```
@@ -0,0 +1,243 @@
+{
+    "TestA": {
+      "name": "Build_Test",
+      "test": "BuildTest",
+      "var1": "/opt/Validation/build.sh",
+      "var2": "/opt/WCSim/install/bin/WCSim",
+      "comment": "Tests defined by suffixed letters were commented out in original tests.txt, they are kept here for comepleteness :)"
+    },
+    "TestB": {
+      "name": "100e_SuperK",
+      "test": "PhysicsValidation",
+      "var1": "/opt/Validation/Generate/Generate.sh",
+      "var2": "/opt/Validation/Generate/macReference/100e_SuperK.mac",
+      "var3": "/opt/Validation/Compare/Reference/analysed_100e_SuperK.root"
+    },
+    "TestC": {
+      "name": "10e_HyperK20pc",
+      "test": "PhysicsValidation",
+      "var1": "/opt/Validation/Generate/Generate.sh",
+      "var2": "/opt/Validation/Generate/macReference/10e_HyperK20pc.mac",
+      "var3": "/opt/Validation/Compare/Reference/analysed_10e_HyperK20pc.root"
+    },
+    "TestD": {
+      "name": "100e_HyperK20pc",
+      "test": "PhysicsValidation",
+      "var1": "/opt/Validation/Generate/Generate.sh",
+      "var2": "/opt/Validation/Generate/macReference/100e_HyperK20pc.mac",
+      "var3": "/opt/Validation/Compare/Reference/analysed_100e_HyperK20pc.root"
+    },
+    "TestE": {
+      "name": "500e_HyperK20pc",
+      "test": "PhysicsValidation",
+      "var1": "/opt/Validation/Generate/Generate.sh",
+      "var2": "/opt/Validation/Generate/macReference/500e_HyperK20pc.mac",
+      "var3": "/opt/Validation/Compare/Reference/analysed_500e_HyperK20pc.root"
+    },
+    "TestF": {
+      "name": "200mu_HyperK20pc",
+      "test": "PhysicsValidation",
+      "var1": "/opt/Validation/Generate/Generate.sh",
+      "var2": "/opt/Validation/Generate/macReference/200mu_HyperK20pc.mac",
+      "var3": "/opt/Validation/Compare/Reference/analysed_200mu_HyperK20pc.root"
+    },
+    "TestG": {
+      "name": "800mu_HyperK20pc",
+      "test": "PhysicsValidation",
+      "var1": "/opt/Validation/Generate/Generate.sh",
+      "var2": "/opt/Validation/Generate/macReference/800mu_HyperK20pc.mac",
+      "var3": "/opt/Validation/Compare/Reference/analysed_800mu_HyperK20pc.root"
+    },
+    "TestH": {
+      "name": "100pi0_HyperK20pc",
+      "test": "PhysicsValidation",
+      "var1": "/opt/Validation/Generate/Generate.sh",
+      "var2": "/opt/Validation/Generate/macReference/100pi0_HyperK20pc.mac",
+      "var3": "/opt/Validation/Compare/Reference/analysed_100pi0_HyperK20pc.root"
+    },
+    "Test1": {
+      "name": "TEST_SK_5MeVem_centre_xdir",
+      "test": "PhysicsValidation",
+      "var1": "Generate/Generate.sh",
+      "var2": "wcsim_test_sk_seed20230628.mac",
+      "var3": "wcsim_test",
+      "var4": "geofile_SuperK.txt"

```

I like the simplicity in the python code that these are extracted in a loop

But it could be better to be more user friendly here by giving the arguments human readable names. Something along the lines of

```suggestion
      "ScriptName": "Generate/Generate.sh",
      "WCSimMacName": "wcsim_test_sk_seed20230628.mac",
      "FileTag": "wcsim_test",
      "GeoFileName": "geofile_SuperK.txt"
```

What do you think?

---

I agree this would be much nicer! Just thinking about it too, if this is going to be generalised, should also account for the case where the variables aren't always called 'Var1' 'Var2' etc... in the config files, will change!